ledger/src/advance.rs

```diff
@@ -22,6 +22,7 @@ impl<N: Network, C: ConsensusStorage<N>> Ledger<N, C> {
         candidate_transactions: Vec<Transaction<N>>,
         candidate_solutions: Option<Vec<ProverSolution<N>>>,
         rng: &mut R,
+        median_timestamp: i64,
```

Comment:
Perhaps rename this as `block_timestamp`, `next_timestamp` or just `timestamp`?  Talking about median timestamp here leaks unnecessary details of the current implementation.